src/pubtools/_pulp/tasks/delete.py

```diff
@@ -564,26 +577,39 @@ def map_to_repo(self, units, repos, unit_attr):
         repo_map = {}
         unit_map = {}
         repos = sorted(repos)
-
-        for unit in sorted(units):
-            unit_name = getattr(unit, unit_attr)
-            unit_map.setdefault(unit_name, RemoveUnitItem(unit=unit, repos=[]))
-            for repo in repos:
-                if repo not in unit.repository_memberships:
-                    LOG.warning(
-                        "%s is not present in %s",
-                        unit_name,
-                        repo,
-                    )
-                else:
+        if repos:
+            for unit in sorted(units):
+                unit_name = getattr(unit, unit_attr)
+                unit_map.setdefault(unit_name, RemoveUnitItem(unit=unit, repos=[]))
+                for repo in repos:
+                    if repo not in unit.repository_memberships:
+                        LOG.warning(
+                            "%s is not present in %s",
+                            unit_name,
+                            repo,
+                        )
+                    else:
+                        repo_map.setdefault(repo, []).append(unit)
+                        unit_map.get(unit_name).repos.append(repo)
+
+            missing = set(repos) - set(repo_map.keys())
+            if missing:
+                missing = ", ".join(sorted(list(missing)))
+                LOG.warning("No units to remove from %s", missing)
+        else:
+            LOG.info("Get repos from units")
+            LOG.info(list(units))
+            for unit in sorted(units):
+                unit_name = getattr(unit, unit_attr)
+                unit_map.setdefault(unit_name, RemoveUnitItem(unit=unit, repos=[]))
+                for repo in unit.repository_memberships:
+                    LOG.info(repo)
+                    if re.match("all-rpm-content-.*", repo):
```

Comment:
Not sure if we need to do this. If user runs delete command for `--all-repos`, they'll probably want to delete it really from all repos.

Reply:
AFAIR packages were usually deleted from the associated repos that were published to the users, unless explicitly defined to clear them from all repos including all-rpm-content.

As discussed earlier, I think it would be good to keep it this way. However, the user should be able to remove the content from all-rpm-content when provided with --repo. 

Wrt to the above comment to get rid of this else block and determine the repos from the membership if none provided, this case to skip the all-rpm-content with --all-repos and include when provided with --repo might look a bit messy. Hence, probably relengs could provide a better insight on the heuristics whether to include all-rpm-content in --all-repos.

Also, "all-rpm-content.*" instead of "all-rpm-content-.* , otherwise it will skip "all-rpm-content" repo.

Reply:
@MichalHaluza ^

Reply:
I'm in favor of keeping this logic and not removing it from `all-rpm-content-.*` even with `--all-repos`.

Existing "delete" commands only have the postcondition that a given piece of content becomes inaccessible from the customer's point of view, not necessarily implying that content has been truly deleted from the remote services.  Deleting from `all-rpm-content-.*` definitely isn't needed in order to achieve that postcondition.